### PR TITLE
Add partsegcore-compiled-backend recipe

### PR DESCRIPTION
**Pull request by goanpeca**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by goanpeca** [comment]
We are in the process of creating conda packages for [napari plugins](https://www.napari-hub.org/) and related dependencies. Once this recipe passes all the checks, every time a new release goes into PyPI, a PR will be automatically open, with usually very minor need for changes so the process of maintaining this moving forward is small. Also the @conda-forge/napari team will be available to help as needed.

You can read a bit more about conda-forge at https://conda-forge.org/#contribute

--- 

@Czaki do you agree with being added as a maintainer?

**Comment by Czaki** [comment]
> @Czaki do you agree with being added as a maintainer?

Yes

**Comment by Czaki** [comment]
Part of the code is C++. And looking on the log the c++ compiler is also need

**Comment by goanpeca** [comment]
>Part of the code is C++. And looking on the log the c++ compiler is also need

Recipe is in draft.

File: recipes/partsegcore-compiled-backend/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    - pip list
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://4dnucleome.cent.uw.edu.pl/PartSeg/
+  summary: PartSegCore compiled backend,
+  license: BSD-3-Clause
+  license_file: License.txt
```
Comment (jaimergp) [inline recipes/partsegcore-compiled-backend/meta.yaml:42]:
```suggestion
  license_file: License.txt
  dev_url: https://github.com/4DNucleome/PartSeg
```

File: recipes/partsegcore-compiled-backend/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - PartSegCore_compiled_backend
+  commands:
+    - pip list
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://4dnucleome.cent.uw.edu.pl/PartSeg/
+  summary: PartSegCore compiled backend,
```
Comment (jaimergp) [inline recipes/partsegcore-compiled-backend/meta.yaml:40]:
```suggestion
  summary: PartSeg is a GUI and a library for segmentation algorithms. This is PartSegCore, its compiled backend.
```

File: recipes/partsegcore-compiled-backend/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - pip
+    - python
+    - setuptools-scm >=3.4
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
+
+test:
+  imports:
+    - PartSegCore_compiled_backend
```
Comment (jaimergp) [inline recipes/partsegcore-compiled-backend/meta.yaml:31]:
```suggestion
    - PartSegCore_compiled_backend
    - PartSegCore_compiled_backend.multiscale_opening
    - PartSegCore_compiled_backend.color_image_cython
    - PartSegCore_compiled_backend.utils
    - PartSegCore_compiled_backend.sprawl_utils.find_split
```

**Pull request merged**